mtcpclient/groupofconnections.go

```diff
@@ -2,15 +2,24 @@ package mtcpclient
 
 import (
 	"fmt"
+	"strconv"
+	"time"
 
 	"github.com/dachad/tcpgoon/tcpclient"
 )
 
-type GroupOfConnections []tcpclient.Connection
+type GroupOfConnections struct {
+	connections []tcpclient.Connection
+	metrics     gcMetrics
```

Comment:
why do you think grouping together the struct representing a group of connections to metrics associated is actually needed?

Reply:
@dcaba , this is the hot topic, because is the reason I needed to pass as a reference to update the metrics.
Basically, my point here was that we needed some kind of global metrics for all the connections (maxConcurrentEstablished), and probably some others. And because of this, the GroupOfConnections was no longer a simple list of Connections. would you get this metric another way simpler than my approach?

Reply:
we had a chat in parallel, and given extra metrics may have sense in the future, let's keep this here.